mlflow/metrics/base.py

```diff
@@ -14,3 +21,306 @@ class MetricValue:
+            )
+            return
+
+        y_pred = eval_df["prediction"]
+        predictions = y_pred.squeeze() if isinstance(y_pred, pd.DataFrame) else y_pred
+        references = eval_df["target"]
+
+        scores = rouge.compute(
+            predictions=predictions,
+            references=references,
+            rouge_types=["rouge2"],
+            use_aggregator=False,
+        )
+        aggregates = rouge.compute(
+            predictions=predictions,
+            references=references,
+            rouge_types=["rouge2"],
+            use_aggregator=True,
+        )
+        return MetricValue(scores=scores["rouge2"], aggregate_results={"": aggregates["rouge2"]})
+
+
+def _rougeL_eval_fn(eval_df, metrics):
+    if "target" in eval_df:
+        try:
+            import evaluate
+
+            rouge = evaluate.load("rouge")
+        except Exception as e:
+            _logger.warning(
+                f"Failed to load 'rouge' metric (error: {e!r}), skipping metric logging."
+            )
+            return
+
+        y_pred = eval_df["prediction"]
+        predictions = y_pred.squeeze() if isinstance(y_pred, pd.DataFrame) else y_pred
+        references = eval_df["target"]
+
+        scores = rouge.compute(
+            predictions=predictions,
+            references=references,
+            rouge_types=["rougeL"],
+            use_aggregator=False,
+        )
+        aggregates = rouge.compute(
+            predictions=predictions,
+            references=references,
+            rouge_types=["rougeL"],
+            use_aggregator=True,
+        )
+        return MetricValue(scores=scores["rougeL"], aggregate_results={"": aggregates["rougeL"]})
+
+
+def _rougeLsum_eval_fn(eval_df, metrics):
+    if "target" in eval_df:
+        try:
+            import evaluate
+
+            rouge = evaluate.load("rouge")
+        except Exception as e:
+            _logger.warning(
+                f"Failed to load 'rouge' metric (error: {e!r}), skipping metric logging."
+            )
+            return
+
+        y_pred = eval_df["prediction"]
+        predictions = y_pred.squeeze() if isinstance(y_pred, pd.DataFrame) else y_pred
+        references = eval_df["target"]
+
+        scores = rouge.compute(
+            predictions=predictions,
+            references=references,
+            rouge_types=["rougeLsum"],
+            use_aggregator=False,
+        )
+        aggregates = rouge.compute(
+            predictions=predictions,
+            references=references,
+            rouge_types=["rougeLsum"],
+            use_aggregator=True,
+        )
+        return MetricValue(
+            scores=scores["rougeLsum"], aggregate_results={"": aggregates["rougeLsum"]}
+        )
+
+
+# general text metrics
+toxicity = make_metric(
+    eval_fn=_toxicity_eval_fn,
+    greater_is_better=False,
+    name="toxicity",
+    long_name="toxicity/roberta-hate-speech-dynabench-r4",
+    version="v1",
+)
```

Comment:
Can we add docstrings for each of these?

Reply:
I've updated the documentation but since I could only get the documentation to work by putting it directly in the .rst file, the docstrings are duplicates and modifying them won't actually change the docs built - should I include a note here to modify documentation in the .rst file or remove the docstrings entirely?